src/Dependencies/Collections/SegmentedDictionary`2+PrivateMarshal.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.Collections.Internal;
+
+namespace Microsoft.CodeAnalysis.Collections;
+
+internal sealed partial class SegmentedDictionary<TKey, TValue>
+{
+    /// <summary>
+    /// Private helper class for use only by <see cref="SegmentedCollectionsMarshal"/>.
+    /// </summary>
+    internal static class PrivateMarshal
+    {
+        /// <inheritdoc cref="SegmentedCollectionsMarshal.GetValueRefOrNullRef{TKey, TValue}(SegmentedDictionary{TKey, TValue}, TKey)"/>
+        public static ref TValue FindValue(SegmentedDictionary<TKey, TValue> dictionary, TKey key)
+            => ref dictionary.FindValue(key);
+
+        /// <inheritdoc cref="SegmentedCollectionsMarshal.GetValueRefOrAddDefault{TKey, TValue}(SegmentedDictionary{TKey, TValue}, TKey, out bool)"/>
+        /// <remarks>
+        /// ⚠️ This method is mirrored by <see cref="SegmentedDictionary{TKey, TValue}.TryInsert(TKey, TValue, InsertionBehavior)"/>.
+        /// If you make any changes here, make sure to keep that version in sync as well.
+        /// </remarks>
+        public static ref TValue? GetValueRefOrAddDefault(SegmentedDictionary<TKey, TValue> dictionary, TKey key, out bool exists)
```

Comment:
>GetValueRefOrAddDefault [](http://example.com/codeflow?start=34&length=23)

If I don't need this right now, do you know of other code that does? It's a fair amount of code if we don't have a known use for it.

Reply:
➡️ Removing from this branch, and will create a different branch to add it if we need it in the future